### PR TITLE
Addition to Datastore and new User class

### PR DESCRIPTION
**Pull request by imanityson**
Created a User class and added a function to the datastore class to store the user and email address. Also made so that the user can add content to the about me section.

### CODE REVIEW

**Review by kassidywebb — CHANGES_REQUESTED** [review]
The code is great & all the functions are named/explained with above comments, nice job! One small change: make sure to name your pull request something along the lines of "Added (what functions or general stuff you added) to (filename).html"

**Comment by vado21** [comment]
Good work on the Datastore creating specific querys when you store a new user and I like that you create a User(Datastore).java to be more organized.

**Review by vado21 — APPROVED** [review]

**Comment by Ajide49** [comment]
Hey Imani, nice job with all the code you added to your AboutMe project. I really like how clean your code is and how the indentation is. I have no objection to your pull request other than changing the name of it to be more descriptive. Nice job!

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -40,6 +40,32 @@ public Datastore() {
 		datastore = DatastoreServiceFactory.getDatastoreService();
 	}
 
+	/** Stores the User in Datastore. */
+	public void storeUser(User user) {
+		Entity userEntity = new Entity("User",user.getEmail());
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/data/Datastore.java:45]: nit: use spaces after commas in general.

Comment (ryanluo) [inline src/main/java/com/google/codeu/data/Datastore.java:45]: nit: use space after commas

Comment (ryanluo) [reply src/main/java/com/google/codeu/data/Datastore.java:45]: Can you also comment on why you initialized the Entity with ("User", user.getEmail())?

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -40,6 +40,32 @@ public Datastore() {
...
+
+		Query query = new Query("User").setFilter(new Query.FilterPredicate("email",FilterOperator.EQUAL, email));
+		PreparedQuery results = datastore.prepare(query);
+		Entity userEntity = results.asSingleEntity();
+
+		if(userEntity == null) {
+			return null;
+		}
+
+	String aboutMe = (String)userEntity.getProperty("aboutMe");
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/data/Datastore.java:64]: nit: space between (String) userEntity

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 3]
```diff
@@ -40,6 +40,32 @@ public Datastore() {
...
+	User user = new User(email, aboutMe);
+
+	return user;
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/data/Datastore.java:67]: return new User(email, aboutMe); ?

File: src/main/java/com/google/codeu/data/User.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+package com.google.codeu.data;
+
+public class User {
+
+  private String email;
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/data/User.java:5]: Maybe make a comment on whether or not you're validating this input

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java [hunk 5]
```diff
@@ -40,9 +43,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
    return;
   }
   
-  String aboutMe = "This is " + user + "'s about me.";
+  User userData = datastore.getUser(user);
+
+  if(userData == null || userData.getAboutMe() == null){
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/servlets/AboutMeServlet.java:48]: nit: spacing
if (userData == null || userData.getAboutMe() == null) {

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 6]
```diff
@@ -40,6 +40,32 @@ public Datastore() {
...
+		userEntity.setProperty("aboutMe",user.getAboutMe());
+		datastore.put(userEntity);
+	}
+	/**
+	* Returns the User owned by the email address, 
+	*null if no matching User was found
+	*/
+	public User getUser(String email) {
+
+		Query query = new Query("User").setFilter(new Query.FilterPredicate("email",FilterOperator.EQUAL, email));
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/data/Datastore.java:56]: If this line is too long (>80 char), split it after setFilter(

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java [hunk 7]
```diff
@@ -40,9 +43,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
+    return;
+  }
   
-  response.getOutputStream().println(aboutMe);
+  response.getOutputStream().println(userData.getAboutMe());
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/servlets/AboutMeServlet.java:52]: Is there a way to populate this in the response without just printing to console?

File: src/main/java/com/google/codeu/servlets/AboutMeServlet.java [hunk 8]
```diff
@@ -56,9 +63,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
   }
   
   String userEmail = userService.getCurrentUser().getEmail();
+  String aboutMe = request.getParameter("about-me");
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/servlets/AboutMeServlet.java:66]: Can this be null? Can we add a comment as to how this parameter is set?

File: target/codeu-starter-project-0.0.1-SNAPSHOT/js/user-page-loader.js [hunk 9]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-function fetchAboutME(){
+function fetchAboutMe(){
   const url = '/about?user=' + parameterUsername;
```
Comment (ryanluo) [inline target/codeu-starter-project-0.0.1-SNAPSHOT/js/user-page-loader.js:18]: QQ: Where is this defined?

File: target/codeu-starter-project-0.0.1-SNAPSHOT/js/user-page-loader.js [hunk 10]
```diff
@@ -25,7 +25,7 @@ function fetchAboutME(){
     }
 
     aboutMeContainer.innerHTML = aboutMe;
-    
+
```
Comment (ryanluo) [inline target/codeu-starter-project-0.0.1-SNAPSHOT/js/user-page-loader.js:28]: nit: delete newline

**Review by ryanluo — COMMENTED** [review]
Really nice :). Mostly nits (small comments)

**Pull request merged**